linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/Trees.scala

```diff
@@ -127,16 +127,19 @@ object Trees {
   }
 
   object Block {
-    def apply(stats: List[Tree])(implicit pos: Position): Tree = {
-      val flattenedStats = stats flatMap {
-        case Skip() => Nil
+    def apply(stats: List[Tree])(implicit pos: Position): Tree =
+      apply(stats.iterator)
+
+    def apply(stats: Iterator[Tree])(implicit pos: Position): Tree = {
+      val flattenedStats = stats.flatMap {
```

Comment:
The implementation of `flatMap` in `Iterator` is significantly more complex than in `List`. I don't think we should lose the fast implementation when the input is known to be a `List`.